sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/SerializableConfiguration.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.hdfs;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * A {@code SerializableConfiguration} allows for serialization of Hadoop's {@code Configuration}.
+ */
+public class SerializableConfiguration implements Externalizable {
```

Comment:
Both are pretty much the same, in the `SerializableConfiguration`, `conf`, given the `Externalizable ` implementation, effectively is "transient". That said thanks for the spark reference tho, we might change the `Configuration` constructor to not load default resources in the `readExternal`, it might be a bit more consistent.

Does it answer your question?